Add experimentalReactChildren option to React integration (#8082)
withastro/astro: .changeset/yellow-snakes-jam.md

---
'@astrojs/react': minor
---

> Optionally parse React slots as React children
> 
> This adds a new flag `experimentalReactChildren` which can be set in the React component like so:

Reviewer: ```suggestion
Optionally parse React slots as React children.

This adds a new configuration option for the React integration `experimentalReactChildren`:
```
